Secciones #84, #85, #86, #87, #88, #89, #90, #91 y #92 (Completas) - Proyecto DevWebCamp (Partes 4 - 7)

CarlosJCD opened the pull request:
Avances en codigo de las secciones #84, #85, #86, #87, #88, #89, #90, #91 y #92 del curso

CarlosJCD committed:
Video #705 - Seccion #84 (INICIO)

- Leyendo los tags escritos en el input de Areas de Experiencia

CarlosJCD committed:
Video #706 - Seccion #84

- Desplegando los tags en el formulario
- Añadiendo la funcionalidad de eliminar un tag con doble click

CarlosJCD committed:
Video #706 - Seccion #84 (FIN)

- Aplicando los estilos a los tags de tecnologias

CarlosJCD committed:
Video #707 - Seccion #85 (INICIO)

- Creacion tabla ponentes en la base de datos
- Creacion modelo de Ponente
- Validacion del formulario de crear ponente

CarlosJCD committed:
Video #708 - Seccion #85

- Leyendo la imagen del formulario de ponentes.
- Generando las versiones webp y png de la imagen del ponente.
- Generando un nuevo hash como nombre de la imagen que se va a guardar.

CarlosJCD committed:
Video #709 - Seccion #85 (FIN)

- Almacenando la imagen del ponente
- Almacenando la informacion del ponente en la base de datos

CarlosJCD committed:
Video #710 - Seccion #86 (INICIO)

- Importando la informacion de los ponentes a la base de datos
- Desplegando los ponentes ya registrados en la pagina de /admin/ponentes

CarlosJCD committed:
Video #711 - Seccion #86

Aplicando estilos a la tabla de ponentes

CarlosJCD committed:
Video #712 - Seccion #86

- Creando la vista de editar Ponente
- Validando el id del url de editar ponente
- Validando que exista un ponente asociado a ese id
- Llenando los campos del formulario con la informacion del ponente

CarlosJCD committed:
Video #713 - Seccion #86

Desplegando la imagen actual del ponente en el formulario de Editar Ponente

CarlosJCD committed:
Video #714 - Seccion #86

- Desplegando los tags del ponente por medio de JS

CarlosJCD committed:
Video #715 - Seccion #86

- Llenando el apartado de redes con la informacion del ponente

CarlosJCD committed:
Video #716 - Seccion #86

- Validando que se haya puesto una imagen nueva
- Asignando la imagen nueva en caso de que haya sido puesta en el formulario
- Almacenando los datos actualizados del ponente en la base de datos

CarlosJCD committed:
Video #717 - Seccion #86

- Ruteo al metodo de eliminar ponentes 
- Pasando el id del ponente a eliminar a traves de un input hidden
- Eliminando el ponente de la base de datos

CarlosJCD committed:
Video #718 - Seccion #86 (FIN)

- Validando si el acceso a una ruta es a traves de un usuario autenticadado y que se admin
- Restringiendo acceso al panel de ponentes a todos los usuarios que no sean admin.

CarlosJCD committed:
Videos #719 y #720 - Seccion #87 (INICIO)

- Introduccion a paginacion
- Creando una clase para la paginacion

CarlosJCD committed:
Video #721 - Seccion #87

- Obteniendo la pagina actual de la paginacion en /admin/ponentes

CarlosJCD committed:
Video #722 - Seccion #87

- Creando un metodo en el modelo de ActiveRecord para contar el total de registros.
- Obteniendo el total de registros de ponentes y asociandolo a la clase Paginacion

CarlosJCD committed:
Video #723 - Seccion #87

Creando metodos en la clase de Paginacion para:
 - Obtener el offset de la pagina actual
 - Obtener el total de paginas
 - Obtener la pagina anterior
 - Obtener la pagina siguiente

CarlosJCD committed:
Video #724 - Seccion #87

- Creando un metodo en la clase Paginacion que genera el enlace a la pagina anterior.
- Creando un metodo en la clase Paginacion que genera el enlace a la pagina siguiente
- Creando un metodo en la clase de Paginacion que genera la seccion de paginacion.

CarlosJCD committed:
Video #725 - Seccion #87

- Generando un metodo en el modelo ActiveRecord que consulta los registros segun un limite y un offset.
- Desplegando los registros segun la paginacion.

CarlosJCD committed:
Video #726 - Seccion #87

Aplicando estilos a la paginacion

CarlosJCD committed:
Video #727 - Seccion #87 (FIN)

- Creando un metodo en la clase Paginacion, generando la paginacion por numeros.
- Aplicando estilos a los numeros de la paginacion

CarlosJCD committed:
Video #728 - Seccion #88 (INICIO)

- Ruteo al metodo de crear evento
- Creacion del archivo /eventos/crear.php 
- Creacion de los campos nombre y descripcion del formulario

CarlosJCD committed:
Video #729 - Seccion #88

- Creacion de la tabla 'categorias' en la base de datos
- Insercion de las categorias "Workshops" y "Conferencias" en la base de datos.
- Desplegando las categorias en el formulario de crear evento

CarlosJCD committed:
Video #730 - Seccion #88

- Creacion de la tabla de dias del evento en la base de datos
- Creando un modelo para los dias del evento
- Desplegando los dias del evento en el formulario de crear evento

CarlosJCD committed:
Video #731 - Seccion #88

- Creacion de la tabla 'horas' en la base de datos
- Registro de las horas en la base de datos
- Creacion de una clase modelo para las horas en la base de datos
- Deplegando las horas registradas en la base de datos en el formulario de crear evento
- Aplicando estilos a las horas desplegadas

CarlosJCD committed:
Video #732 - Seccion #88

Creando campo para el ponente del evento
Creando campo para la cantidad de lugares en el evento

CarlosJCD committed:
Video #733 - Seccion #88

Creacion de la tabla eventos en la base de datos.
Creacion del modelo Evento.
Creacion de un metodo de validacion para evento

CarlosJCD committed:
Video #734 - Seccion #88

- Validando los campos del formulario de crear evento
- Manteniendo los valores de los campos del formulario de  crear evento

CarlosJCD committed:
Video #735 - Seccion #88

Almacenando los cambios en las selecciones de los dias y categoria del evento por medio de JS

CarlosJCD committed:
Video #736 - Seccion #88

- Creando un API endpoint para consultar el horario asociado al dia
- Validando los datos que recibe el servidor
- Creando una clase modelo para el horario asociado al dia

CarlosJCD committed:
Video #737 - Seccion #88

Creando metodo en la clase ActiveRecord para consultar por medio de un array la base de datos.
Mandando una respuesta con los resultados en forma de json

CarlosJCD committed:
Video #738 - Seccion #88

CarlosJCD committed:
Video #739 - Seccion #88

CarlosJCD committed:
Video #740 - Seccion #88

CarlosJCD committed:
Video #741 - Seccion #88

CarlosJCD committed:
Video #742 - Seccion #88

CarlosJCD committed:
Video #743 - Seccion #88

CarlosJCD committed:
Video #744 - Seccion #88

CarlosJCD committed:
Video #745 - Seccion #88

CarlosJCD committed:
Video #746 - Seccion #88

CarlosJCD committed:
Video #747 - Seccion #88

CarlosJCD committed:
Video #748 - Seccion #88 (FIN)

CarlosJCD committed:
Video #749 - Seccion #89 (INICIO)

CarlosJCD committed:
Videos #750 y #751 - Seccion #89

CarlosJCD committed:
Video #752 - Seccion #89

CarlosJCD committed:
Video #753 - Seccion #89

CarlosJCD committed:
Video #754 - Seccion #89 (FIN)

CarlosJCD committed:
Video #755 y #756 - Seccion #90 (INICIO)

CarlosJCD committed:
Video #757 - Seccion #90

CarlosJCD committed:
Video #758 - Seccion #90

CarlosJCD committed:
Video #759 - Seccion #90

CarlosJCD committed:
Video #760 - Seccion #90

CarlosJCD committed:
Video #761 - Seccion #90

CarlosJCD committed:
Video #762 y #763 - Seccion #90

CarlosJCD committed:
Video #764 - Seccion #90

CarlosJCD committed:
Video #765 - Seccion #90

CarlosJCD committed:
Video #766 - Seccion #90 (FIN)

CarlosJCD committed:
Video #767 - Seccion #91 (INICIO)

CarlosJCD committed:
Video #768 - Seccion #91

CarlosJCD committed:
Video #769 y #770 - Seccion #91

CarlosJCD committed:
Video #771 - Seccion #91

CarlosJCD committed:
Video #772 - Seccion #91

CarlosJCD committed:
Videos #773 y #774 - Seccion #91

CarlosJCD committed:
Video #775 - Seccion #91 (FIN)

CarlosJCD committed:
Videos #776 y #777 - Seccion #92 (INICIO)

CarlosJCD committed:
Video #778 - Seccion #92

CarlosJCD committed:
Video #779 - Seccion #92

CarlosJCD committed:
Video #780 - Seccion #92

CarlosJCD committed:
Video #781 - Seccion #92

CarlosJCD committed:
Video #782 - Seccion #92

CarlosJCD committed:
Video #783 - Seccion #92

jfer1990 reviewed (changes requested):
Carlos, hay comentarios de este pr en discord, chécalos y me comentas porfavor

CarlosJCD committed:
Correcciones imagenes y overflow

CarlosJCD committed:
Correcciones vistas de Ponentes y eventos

CarlosJCD committed:
Ajuste al crear cuenta

Ajuste al proceso de crear cuenta, donde primero se redirige al usuario a la ruta /mensaje y luego se envia el email de confirmacion.
